### PR TITLE
BROC-436: Add Statuspage as a provider

### PR DESCRIPTION
**Pull request by Psuedoo**
With this you can add Statuspage as an impact source

### CODE REVIEW

**Comment by Psuedoo** [comment]
What am I missing here? I have this in my tf plan:
```tf
resource "sleuth_incident_impact_source" "statuspage" {
  project_slug     = "myorg-app"
  environment_name = "production"
  name             = "statuspage babyyyy"
  provider_name    = "STATUSPAGE"
  statuspage_input {
    remote_page = "gg90hvtxx5ns"
  }

}
```

And I'm getting this error:
```
│ Error: Unsupported block type
│ 
│   on main.tf line 20, in resource "sleuth_incident_impact_source" "statuspage":
│   20:   statuspage_input {
│ 
│ Blocks of type "statuspage_input" are not expected here.
```

**Comment by jrgarcia** [comment]
@Psuedoo `statuspage_input` was added in this commit. The only thing I can think of is that the binaries either didn't get compiled or it's looking for them in the wrong place.

File: internal/gqlclient/models.go [hunk 1]
```diff
@@ -240,11 +240,19 @@ type BlamelessProviderData struct {
 	RemoteSeverityThreshold string   `json:"remoteSeverityThreshold"`
 }
 
+type StatuspageProviderData struct {
+	RemotePage                 string `json:"remotePage"`
+	RemoteComponent            string `json:"remoteComponent"`
+	RemoteImpact               string `json:"remoteImpact"`
+	IgnoreMaintenanceIncidents bool   `json:"ignoreMaintenceIncidents"`
```
Comment (jrgarcia) [inline internal/gqlclient/models.go:247]:
```suggestion
	IgnoreMaintenanceIncidents bool   `json:"ignoreMaintenanceIncidents"`
```

**Review by jrgarcia — APPROVED** [review]
Works just fine after making the changes I added.

**Pull request merged**